pkg/pillar/pubsub/publish.go

```diff
@@ -266,24 +266,41 @@ func (pub *PublicationImpl) populate() {
 		return
 	}
 	for key, itemB := range pairs {
-		// Just in case large items were stored separately
-		itemB, err = readAddLarge(pub.log, itemB)
-		if err != nil {
-			// Handle missing files??
-			pub.log.Error(err)
-		}
-		item, err := parseTemplate(pub.log, itemB, pub.topicType)
-		if err != nil {
-			// Handle bad files such as those of size zero
-			pub.log.Error(err)
-			continue
+		err := pub.populateItem(key, itemB)
+		if err != nil && pub.persistent {
+			// Item data are possibly corrupted, try to populate from backup.
+			itemB, err = pub.driver.RecoverFromBackup(key)
+			if err == nil {
+				err = pub.populateItem(key, itemB)
+			}
```

Comment:
With this code in place why do we also need recovery code in the Load() function?

Reply:
Done: moved all recovery code to `Load`.